datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java

```diff
@@ -270,6 +260,22 @@ public void clear() {
 
   }
 
+  @Override public void deleteDatamapData(Segment segment) {
+    try {
+      String segmentId = segment.getSegmentNo();
+      String datamapPath = CarbonTablePath
+          .getDataMapStorePath(tableIdentifier.getTablePath(), segmentId, dataMapName);
+      if (FileFactory.isFileExist(datamapPath)) {
+        CarbonFile file = FileFactory.getCarbonFile(datamapPath,
+            FileFactory.getFileType(datamapPath));
+        CarbonUtil.deleteFoldersAndFilesSilent(file);
+      }
+    } catch (IOException | InterruptedException ex) {
+      throw new RuntimeException(
```

Comment:
Seems throw IOException is better, then catch it and log it in out most caller

Reply:
please do the same for line 279

Reply:
OK